torchrl/objectives/a2c.py

```diff
@@ -445,7 +475,26 @@ def loss_critic(self, tensordict: TensorDictBase) -> torch.Tensor:
                 f"TDLambdaEstimate and TDEstimate all return a 'value_target' entry that "
                 f"can be used for the value loss."
             )
-        return self.critic_coef * loss_value
+        clip_fraction = None
+        if self.clip_value:
+            clip_value = self.clip_value.to(state_value.device)
+            state_value_clipped = old_state_value + (
+                state_value - old_state_value
+            ).clamp(-clip_value, clip_value)
+            loss_value_clipped = distance_loss(
+                target_return,
+                state_value_clipped,
+                loss_function=self.loss_critic_type,
+            )
+            # Chose the most pessimistic value prediction between clipped and non-clipped
+            loss_value = torch.max(loss_value, loss_value_clipped)
+            clip_fraction = (
+                (state_value / old_state_value)
+                .clamp(1 - clip_value, 1 + clip_value)
+                .abs()
+                .detach()
+            )
+        return self.critic_coef * loss_value, clip_fraction
```

Comment:
That seems to be a lot of duplicated code, can we move that to objectives.utils within a private function?